### PR TITLE
feat: api add challenge

### PR DESCRIPTION
**Pull request by Z-a-r-a-k-i**

### CODE REVIEW

File: go/docker-compose.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ services:
     volumes:
       - mysql_data:/var/lib/mysql
     ports:
-      - 3306
+      - 3306:3306
```
Comment (moul) [inline go/docker-compose.yml]: please remove it as soon as you finish your dev

Comment (moul) [reply go/docker-compose.yml]: there is a `make mysql.shell` helper btw

File: go/pkg/pwapi/httpclient.go [hunk 2]
```diff
@@ -42,6 +42,12 @@ func (c HTTPClient) AdminPS(input *AdminPS_Input) (AdminPS_Output, error) {
 	return result, err
 }
 
+func (c HTTPClient) AdminAddChallenge(input *AdminChallengeAdd_Input) (AdminChallengeAdd_Output, error) {
+	var result AdminChallengeAdd_Output
+	err := c.doGet("/admin/add-challenge", input, &result)
```
Comment (moul) [inline go/pkg/pwapi/httpclient.go]:
```suggestion
	err := c.doPost("/admin/add-challenge", input, &result)
```

File: go/docker-compose.yml (hunk 1 above)
Comment (Z-a-r-a-k-i) [reply go/docker-compose.yml]: this is for forestadmin to connect via npm start

Comment (moul) [reply go/docker-compose.yml]: create a complementary docker-compose.yml file that will override this rule if you want, but default docker-compose should expose as less ports as possible

(anyone with a running mysql cannot run this docker-compose file, which is too common)

File: go/pkg/pwapi/api_admin-challenge-add.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	var challenge pwdb.Challenge
+	challenge.Name = in.Name
+	challenge.Description = in.Description
+	challenge.Author = in.Author
+	challenge.Locale = in.Locale
+	challenge.IsDraft = in.IsDraft
+	challenge.PreviewUrl = in.PreviewUrl
+	challenge.Homepage = in.Homepage
+
+	svc.db.Create(&challenge)
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-add.go]: check err

File: go/pkg/pwapi/api_admin-challenge-add.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	}
+
+	var challenge pwdb.Challenge
+	challenge.Name = in.Name
+	challenge.Description = in.Description
+	challenge.Author = in.Author
+	challenge.Locale = in.Locale
+	challenge.IsDraft = in.IsDraft
+	challenge.PreviewUrl = in.PreviewUrl
+	challenge.Homepage = in.Homepage
```
Comment (moul) [inline go/pkg/pwapi/api_admin-challenge-add.go]:
```suggestion
	challenge := pwdb.Challenge{
	    Name: in.Name,
	    Description: in.Description,
	    Author: in.Author,
	    Locale: in.Locale,
	    IsDraft: in.IsDraft,
	    PreviewUrl: in.PreviewUrl,
	    Homepage: in.Homepage,
	}
```

File: api/pwapi.proto [hunk 5]
```diff
@@ -104,6 +105,21 @@ message AdminPS {
...
 
+message AdminChallengeAdd {
+  message Input {
+    string name = 1 [(gogoproto.moretags) = "url:\"name\""];
+    string description = 2 [(gogoproto.moretags) = "url:\"description\""];
+    string author = 3 [(gogoproto.moretags) = "url:\"author\""];
+    string locale = 4 [(gogoproto.moretags) = "url:\"locale\""];
+    bool is_draft = 5 [(gogoproto.moretags) = "url:\"is_draft\""];
+    string preview_url = 6 [(gogoproto.moretags) = "url:\"preview_url\""];
+    string homepage = 7 [(gogoproto.moretags) = "url:\"homepage\""];
```
Comment (moul) [inline api/pwapi.proto]: can be replaced by

```suggestion
    pwdb.Challenge challenge = 1;
```

:)

File: go/docker-compose.yml (hunk 1 above)
Comment (Z-a-r-a-k-i) [reply go/docker-compose.yml]: Okay looks already good in docker-compose.dev.yml, will revert it

**Pull request merged**